sympy/physics/vector/frame.py

```diff
@@ -554,6 +556,25 @@ def _dcm(self, parent, parent_orient):
             self._dcm_dict = self._dlist[0] = {}
         # Reset the _dcm_cache
             self._dcm_cache = {}
+
+        else:
+        #Check for loops and raise warning accordingly.
```

Comment:
Can `topological_sort` be used to detect this? It will raise ValueError if there is a cycle detected.

Reply:
We don't want to give error as user may actually want to establish a new relationship by make frames cyclic.

Plus that's how we did it for point velocity.

Reply:
Couldn't you catch the error and raise the warning ? That way there would be little code added.

Reply:
I don't think catching error for raising warning would be nice as in this way the code would be consistent.

Reply:
I'm just wondering if the code could look like this:
```python
#Check for loops and raise warning accordingly.
try:
    check = topological_sort(...)
except ValueError:
    warn('Loops are defined among the orientation of frames.' + \
    ' This is likely not desired and may cause errors in your calculations.')
```

Reply:
@moorepants What do you think?

Reply:
sorting is probably more than you need since you are only trying to detect a loop, not put them in order

Reply:
I think it should be a warning if it still sets the relationships and an error if it doesn't.